apps/engineeringblog/components/Article/ArticleCard.tsx

```diff
@@ -0,0 +1,62 @@
+import { ArticleWithoutContent } from "@/engineeringblog/utils";
+import {
+  Card,
+  CardActionArea,
+  CardContent,
+  CardMedia,
+  Typography,
+} from "@mui/material";
+import React from "react";
+
+const ArticleCard = React.forwardRef(function ArticleCard(
+  {
+    article,
+  }: {
+    article: ArticleWithoutContent;
+  },
+  ref: React.Ref<HTMLDivElement>,
+) {
+  return (
+    <Card
+      elevation={0}
+      square={true}
+      variant="outlined"
+      sx={{
+        "&:hover": {
+          border: `1px solid "#ED1C24"`,
+          img: {
+            filter: "none",
+          },
+        },
+        border: "1px solid #DAD5D5",
+        filter: "drop-shadow(0px 4px 8px rgba(0, 0, 0, 0.1))",
+      }}
+    >
+      <CardActionArea
+        component={article.slug ? "a" : "div"}
```

Comment:
@kilemensi Because for this app we've used the `app router` directory structure which has [a new way to mount ](https://nextjs.org/docs/messages/next-router-not-mounted#:~:text=This%20can%20also%20happen%20when%20you%20try%20to%20use%20the%20useRouter%20hook%20from%20next/router%20inside%20the%20app%20directory%2C%20as%20the%20App%20Router%27s%20useRouter%20from%20next/navigation%20has%20different%20behavior%20to%20the%20useRouter%20hook%20in%20pages.)the `router`. Using it throws an error about the router not being mounted yet.

Reply:
Cool, but when we come across errors, aren't we supposed to fix 'em?

My issue with `a` is the app will fully reload every time a user click a link.

Reply:
I have used `next/link`. I will update it once #877 is merged.